cp-connector/pkg/nats/nats.go

```diff
@@ -139,6 +139,12 @@ func (nc *NatsConnector) QueueSubscribeMultiple(subjects []string, queueGroup st
 		return ErrSubNilMessageProcessor
 	}
 
+	// Immediately verify if the Connection is valid, to avoid starting go routines for a
+	// faulty nats connection with no subjects
+	if _, err := nc.getOrCreateConnection(); err != nil {
```

Comment:
nitpick: let's just rename this one to `ensureConnection()` i think this is also how it's called in mongodb datastore / shipyard-controller